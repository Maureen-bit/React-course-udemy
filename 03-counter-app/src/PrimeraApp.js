// importar react ayuda a solucionar el error de: "Jsx must be in a react scope"
import React from 'react';
import  PropTypes from 'prop-types';

const PrimeraApp = ( { saludo, subtitulo } ) => {
    const datos = {
        nombre: "Maureen",
        edad: 23
    }

    return (   
        <>
        {/* Como react no deja imprimir objetos directamente porque no es vàlido como hijo, entonces usamos 
        JSON.stringify para convertirlo en string y luego en los argumentos ponemos un espacio de 3 para que lo 
        formatee, dentro de la etiqueta pre */}
        <pre>{ JSON.stringify( datos, null, 3 ) }</pre>
        <h1>{saludo}</h1>
        <h2>{subtitulo}</h2>
        <p>Esta es la primera aplicación del curso</p>
        </>
    )
};

/* Esto se realiza para que la propiedad saludo sea obligatoria y con un valor de tipo string,
de lo contrario saldrá un error o warning en consola */
PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired,
}

/* Este se utiliza cuando quiera poner un valor de alguna propiedad por defecto sin necesidad de esperarla */
PrimeraApp.defaultProps = {
    subtitulo: "Soy un subtitulo"
}
export default PrimeraApp;