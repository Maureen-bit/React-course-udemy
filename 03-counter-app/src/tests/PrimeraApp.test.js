import React from "react";
import { render } from "@testing-library/react";
import PrimeraApp from "../PrimeraApp";
import {shallow} from "enzyme";
import '@testing-library/jest-dom'

describe('Pruebas en componente: PrimeraApp', () => {
  /* Si quiero que esta prueba sirva debo descomentar la linea de 
  import '@testing-library/jest-dom/extend-expect'; para que se 
  extiendan caracteristicas del expect
  
  test('Debe de mostrar el mensaje Hola Maureen', () => {
    const saludo = "Hola Maureen";
    const view = render(<PrimeraApp saludo={saludo}/>);
    // Pregunta si existe el saludo en el documento
    expect(view.getByText(saludo)).toBeInTheDocument();
  }); */

  test('Debe de mostra el componenten <PrimeraApp />> correctamente ', () => {
    const saludo = "Hola Maureen";
    const view = shallow(<PrimeraApp saludo={saludo}/>);
    /* Snapshot le toma una foto instantanea al componente y compara el componente y 
    el snapshot tomado, si no coinciden la prueba falla, esto puede ocurrir por algÃºn cambio
    en el componente original*/
    expect(view).toMatchSnapshot();
  });
  
  test('Debe de mostra el subtitulo enviado por props al componenten <PrimeraApp /> ', () => {
    const saludo = "Hola Maureen";
    const subTitulo = "Soy el subtitulo de las props"
    const view = shallow(<PrimeraApp saludo={saludo} subtitulo={subTitulo}/>);
    /*Retorna el texto que tiene la etiqueta h2, si hay mÃ¡s d euna etiqueta devuelve un arreglo */
    const textoParrafo = view.find('h2').text();
    console.log("ðŸš€ ~ file: PrimeraApp.test.js ~ line 33 ~ test ~ textoParrafo", textoParrafo);
    expect(textoParrafo).toBe(subTitulo);
  });
});
