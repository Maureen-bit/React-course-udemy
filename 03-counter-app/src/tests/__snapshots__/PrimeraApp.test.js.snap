// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente: PrimeraApp Debe de mostra el componenten <PrimeraApp />> correctamente  1`] = `
<Fragment>
  <pre>
    {
   "nombre": "Maureen",
   "edad": 23
}
  </pre>
  <h1>
    Hola Maureen
  </h1>
  <h2>
    Soy un subtitulo
  </h2>
  <p>
    Esta es la primera aplicaci√≥n del curso
  </p>
</Fragment>
`;
